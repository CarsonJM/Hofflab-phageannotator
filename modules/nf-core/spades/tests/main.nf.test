nextflow_process {

    name "Test Process: SPADES"
    script "../main.nf"
    process "SPADES"

    test("fastq.gz paired") {
        when {
            params {
                spades_single_args = "--meta --only-assembler"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                    ],
                    [],
                    []
                ]
                input[1] = []
                input[2] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }


    test("fastq.gz - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                    ],
                    [],
                    []
                ]
                input[1] = []
                input[2] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
