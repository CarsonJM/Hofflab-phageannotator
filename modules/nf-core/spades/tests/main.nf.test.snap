{
    "fastq.gz paired": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "contigs": [
                    
                ],
                "gene_clusters": [
                    
                ],
                "gfa": [
                    
                ],
                "log": [
                    
                ],
                "scaffolds": [
                    
                ],
                "transcripts": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T12:23:03.958235483"
    },
    "fastq.gz - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "contigs": [
                    
                ],
                "gene_clusters": [
                    
                ],
                "gfa": [
                    
                ],
                "log": [
                    
                ],
                "scaffolds": [
                    
                ],
                "transcripts": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-23T12:37:52.485295905"
    }
}